api/uniter/unit.go

```diff
@@ -756,3 +756,26 @@ func (u *Unit) NetworkInfo(bindings []string, relationId *int) (map[string]param
 
 	return results.Results, nil
 }
+
+// Series returns the operating system series running on the unit.
+//
+// NOTE: Unlike state.Unit.Series(), this method returns an error
+// as well, because it needs to do an API call.
+func (u *Unit) Series() (string, error) {
```

Comment:
Please maybe add a TODO - this should be implemented like Life() - an attribute on the Unit which is updated by Refresh(). If you look at how the uniter works, when a unit change event is processed, the first thing done is a Refresh(), and then Life() and Series() and Resolved() are called. Having Series() make a another API call to get something that could be returned via Refresh() is not preferred. The fact that Resolved() is not done this way also is unfortunate, but to fix would require a little facade surgery. Part of me thinks we should make this facade fix - we already have enough issues with the controller being swamped by API calls on models with many units.

Reply:
Per conversation, implementing this change instead of adding a TODO.